diff --git a/dist/index.js b/dist/index.js
--- a/dist/index.js
+++ b/dist/index.js
@@ -549,6 +549,9 @@ async function getDefaultDownloadFetch() {
   return safeNodeFetchPromise != null ? safeNodeFetchPromise : safeNodeFetchPromise = createSafeNodeFetch();
 }
 function isNodeDefaultFetch(fetch) {
+  if (typeof fetch !== "function") {
+    return false;
+  }
   const source = Function.prototype.toString.call(fetch);
   return source.includes("internal/deps/undici") || source.includes("lazy loading of undici");
 }
diff --git a/dist/index.mjs b/dist/index.mjs
--- a/dist/index.mjs
+++ b/dist/index.mjs
@@ -447,6 +447,9 @@ async function getDefaultDownloadFetch() {
   return safeNodeFetchPromise != null ? safeNodeFetchPromise : safeNodeFetchPromise = createSafeNodeFetch();
 }
 function isNodeDefaultFetch(fetch) {
+  if (typeof fetch !== "function") {
+    return false;
+  }
   const source = Function.prototype.toString.call(fetch);
   return source.includes("internal/deps/undici") || source.includes("lazy loading of undici");
 }
